### PR TITLE
Peter Code Updates

### PR DESCRIPTION
**Pull request by rydzynskip**
Many small updates

### CODE REVIEW

File: ArduinoCode/Pedal_Box/Pedal_Box.ino [hunk 1]
```diff
@@ -131,7 +154,8 @@ void readPotentiometers() {
     
     unsigned char message[8] = {accelTorque,regenTorque,0,0,isForward,1,0,0};
     CAN.sendMsgBuf(CAN_MOTOR, 0, 8, message); // send message to motor controller with torques and direction
-  } else {  // turn off motor and send error message
+  } else {  // send shutdown message, turn off motor, and send error message
```
Comment (mccauleyma) [inline ArduinoCode/Pedal_Box/Pedal_Box.ino:157]: Doing BMS first means you'll be cutting off power into the inverter immediately. However given the capacitance of the inverter I theoretically think this should be fine and it'll still get shutdown normally as well anyways. Just an FYI

**Review by mccauleyma — APPROVED** [review]
All looks good to me! Only question: why the move to 250kbps? Just for testing? Or is there a reason?

**Comment by rydzynskip** [comment]
The charger requires the baud rate to be 250 kbps for some reason.

**Comment by mccauleyma** [comment]
Oh jeez, really? We may end up having to use the two CAN connections on the BMS if that becomes an issue. And run just the charger on 250kbps and the rest on 1mbps. I guess for now just be aware of it until we can actually test how saturated the network gets?

File: ArduinoCode/Pedal_Box/Pedal_Box.ino [hunk 2]
```diff
@@ -97,9 +115,13 @@ void readStates() {
       invertorOn = buf[0]; // set inverter state
     } else if (canId == CAN_DIRECTION) {
       isForward = buf[0]; // set direction state
-    } else if (canId = CAN_BMS_STATES) {
+    } else if (canId == CAN_BMS_STATES) {
       dischargeEnabled = buf[5] & 1;     // get first bit of Relay State byte
       chargeEnabled = (buf[5] >> 1) & 1; // get second bit of Relay State byte 
+    } else if (canId == CAN_MC_FAULTS) {
+      if ((buf[4] != 0) || ((buf[5] >> 3) & 1) || ((buf[5] >> 4) & 1)) { // only send message for fault bits 32-39, 43, 44
```
Comment (jamescd18) [inline ArduinoCode/Pedal_Box/Pedal_Box.ino:122]: this kind of code formatting of having multiple or statements in an if condition is kinda tough to read. I would recommend splitting it up in these cases so it looks something more like this, but it's up to you:
```
bool conditionA = buf[4] != 0;
bool conditionB = (buf[5] >> 3) & 1;
bool conditionC = (buf[5] >> 4) & 1;
if (conditionA || conditionB || conditionC) {
  ... insert code ...
}
```

**Review by jamescd18 — APPROVED** [review]
It probably looks good to me from my knowledge of what's going on. One suggestion down below. Also, I strongly strongly strongly strongly recommend making your PRs more frequently so they're smaller. Its just more difficult to read all the changes and stuff the more it is. Like, switching to `CAN_250KBPS` across a bunch of files is definitely good to do in a separate PR.

File: ArduinoCode/Pedal_Box/Pedal_Box.ino (hunk 2 above)
Comment (mccauleyma) [reply ArduinoCode/Pedal_Box/Pedal_Box.ino:122]: Not recommended imo. Remember we are working on a lil Arduino. So now you're making 3 more variables that aren't needed and tbh idk enough about the compiler to recall if it actually simplifies those or not.

**Pull request merged**